intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/parser/context_utils.py

```diff
@@ -77,7 +82,72 @@ def read_pdf(pdf_path):
                     else:
                         pageimg=pageimg+'.'
                 result=result+pageimg
-    return result
+
+    tables_result = []
+    def get_relation(table_coords, caption_coords, table_page_number, caption_page_number, threshold=100):
```

Comment:
it would be better to make this function code is clear or could move into a independent call function